src/lava/magma/core/model/py/ports.py

```diff
@@ -270,30 +593,49 @@ class PyVarPort(AbstractPyPort):
     def __init__(self,
                  var_name: str,
                  csp_send_port: ty.Optional[CspSendPort],
-                 csp_recv_port: ty.Optional[CspRecvPort], *args):
+                 csp_recv_port: ty.Optional[CspRecvPort],
+                 process_model: AbstractProcessModel,
+                 shape: ty.Tuple[int, ...] = tuple(),
+                 d_type: type = int):
         self._csp_recv_port = csp_recv_port
         self._csp_send_port = csp_send_port
         self.var_name = var_name
-        super().__init__(*args)
+        super().__init__(process_model, shape, d_type)
 
     @property
     def csp_ports(self) -> ty.List[AbstractCspPort]:
-        """Returns all csp ports of the port."""
+        """Property to get the corresponding CSP Ports of all connected
+        PyPorts (csp_ports). The CSP Port is the low level interface of the
+        backend messaging infrastructure which is used to send and receive data.
+
+        Returns
+        -------
+        A list of all CSP Ports connected to the PyPort.
+        """
         if self._csp_send_port is not None and self._csp_recv_port is not None:
             return [self._csp_send_port, self._csp_recv_port]
         else:
-            # In this case the port was not connected
+            # In this case the Port was not connected
             return []
 
+    @abstractmethod
     def service(self):
+        """Abstract method to set the value of the related Var of the VarPort,
+        received from the connected RefPort, or to send the value of the related
+        Var of the VarPort to the connected RefPort. The connected RefPort sends
+        a command VarPortCmd first to determine a read() or write() operation.
+        """
```

Comment:
The connected RefPort determines whether it will perform a read() or write() operation by sending a command VarPortCmd.

Reply:
Changed